src/main/java/net/minecraftforge/fluids/FluidUtil.java

```diff
@@ -19,148 +37,453 @@ private FluidUtil()
-        if (liquid != null && liquid.amount > 0)
+    }
+
+    /**
+     * Helper method to get an IFluidHandler for at a block position.
+     *
+     * Returns null if there is no valid fluid handler.
+     */
+    @Nullable
+    public static IFluidHandler getFluidHandler(World world, BlockPos blockPos, @Nullable EnumFacing side)
+    {
+        IBlockState state = world.getBlockState(blockPos);
+        Block block = state.getBlock();
+
+        if (block.hasTileEntity(state))
         {
-            // play sound
-            if(player != null)
+            TileEntity tileEntity = world.getTileEntity(blockPos);
+            if (tileEntity != null && tileEntity.hasCapability(CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY, side))
             {
-                SoundEvent soundevent = liquid.getFluid().getFillSound(liquid);
-                player.playSound(soundevent, 1f, 1f);
+                return tileEntity.getCapability(CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY, side);
             }
+        }
+        else if (block instanceof IFluidBlock)
+        {
+            return new FluidBlockWrapper((IFluidBlock) block, world, blockPos);
+        }
+        else if (block instanceof BlockLiquid)
+        {
+            return new BlockLiquidWrapper((BlockLiquid) block, world, blockPos);
+        }
+
+        return null;
+    }
+
+    /**
+     * Tries to place a fluid in the world in block form.
+     * Makes a fluid emptying sound when successful.
+     * Checks if water-like fluids should vaporize like in the nether.
+     *
+     * Modeled after {@link net.minecraft.item.ItemBucket#tryPlaceContainedLiquid(EntityPlayer, World, BlockPos)}
+     *
+     * @param player  Player who places the fluid. May be null for blocks like dispensers.
+     * @param worldIn World to place the fluid in
+     * @param fluid   The fluid to place.
+     * @param pos     The position in the world to place the fluid block
+     * @return true if successful
+     */
+    public static boolean tryPlaceFluid(@Nullable EntityPlayer player, World worldIn, Fluid fluid, BlockPos pos)
+    {
+        if (worldIn == null || fluid == null || pos == null)
+        {
+            return false;
+        }
+
+        // check that we can place the fluid at the destination
+        IBlockState destBlockState = worldIn.getBlockState(pos);
+        Material destMaterial = destBlockState.getMaterial();
+        boolean isDestNonSolid = !destMaterial.isSolid();
+        boolean isDestReplaceable = destBlockState.getBlock().isReplaceable(worldIn, pos);
+        if (!worldIn.isAirBlock(pos) && !isDestNonSolid && !isDestReplaceable)
+        {
+            return false; // Non-air, solid, unreplacable block. We can't put fluid here.
+        }
+
+        IBlockState fluidBlockState = fluid.getBlock().getDefaultState();
+
+        if (worldIn.provider.doesWaterVaporize() && fluidBlockState.getMaterial() == Material.WATER)
```

Comment:
I actually had that and removed it since I don't want to go of track with this PR. 
It will be easy to add separately. Just need a `fluid.doesVaporize()` and `fluid.vaporize(world, pos, blah)`.